### PR TITLE
improve workflow with fastify-integration job

### PR DESCRIPTION
**Pull request by Uzlopak**
Tested with https://github.com/fastify/light-my-request/pull/269

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .github/workflows/plugins-ci.yml [hunk 1]
```diff
@@ -120,12 +125,47 @@ jobs:
...
+
   automerge:
     name: Automerge Dependabot PRs
     if: >
         github.event_name == 'pull_request' &&
         github.event.pull_request.user.login == 'dependabot[bot]'
-    needs: test
+    needs: 
+      - test
+      - fastify-integration
```
Comment (Eomm) [inline .github/workflows/plugins-ci.yml:168]: is `fastify-integration` considered done when we have `inputs.fastify-integration = false`?

File: .github/workflows/plugins-ci.yml [hunk 2]
```diff
@@ -120,12 +125,47 @@ jobs:
...
+
+      - name: Test Fastify Integration
+        run: |
+          npm link &&
+          PACKAGE_NAME=$(node -p "require('./package.json').name")
+          git clone https://github.com/fastify/fastify.git --depth 1 &&
+          cd fastify &&
+          npm i --ignore-scripts &&
+          npm link $PACKAGE_NAME &&
+          npm run unit
```
Comment (Eomm) [inline .github/workflows/plugins-ci.yml:159]: not sure about this usage/wording:
if we set the `inputs.fastify-integration=true` on a plugin, nothing will happen here if I'm not wrong.
we will run the fastify's test with a `fastify-postgres` plugin linked

I would just ask chatgpt a clearer var name

Maybe: `fastify-library-integration`?

**Comment by Uzlopak** [comment]
@Eomm 

PTAL

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Fdawgs — APPROVED** [review]

**Review by jsumners — APPROVED** [review]

**Pull request merged**